tests/forte/data/extractor/seqtagging_extractor_test.py

```diff
@@ -0,0 +1,70 @@
+#  Copyright 2020 The Forte Authors. All Rights Reserved.
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#       http://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import unittest
+
+from ft.onto.base_ontology import Sentence, Token, EntityMention
+from forte.pipeline import Pipeline
+from forte.data.readers.conll03_reader import CoNLL03Reader
+from forte.data.data_pack import DataPack
+from forte.data.extractor.seqtagging_extractor import BioSeqTaggingExtractor
+
+
+class SeqTaggingExtractorTest(unittest.TestCase):
+
+    def setUp(self):
+        # Define and config the Pipeline
+        self.dataset_path = "data_samples/conll03"
+
+    def test_BioSeqTaggingExtractor(self):
+        pipeline = Pipeline[DataPack]()
+        reader = CoNLL03Reader()
+        pipeline.set_reader(reader)
+        pipeline.initialize()
+
+        config = {
+            "entry_type": EntityMention,
+            "need_pad": True,
+            "attribute": "ner_type",
+            "based_on": Token,
+        }
+
+        expected = [(None, 'O'), ('ORG', 'B'), ('ORG', 'I'),
+                    (None, 'O'), (None, 'O'), (None, 'O'),
+                    (None, 'O'), (None, 'O'), (None, 'O'),
+                    ('MISC', 'B'), (None, 'O'), (None, 'O'),
+                    (None, 'O'), (None, 'O'), (None, 'O'),
+                    ('MISC', 'B'), (None, 'O'), (None, 'O'),
+                    (None, 'O'), (None, 'O'), (None, 'O'),
+                    (None, 'O'), (None, 'O'), (None, 'O'),
+                    (None, 'O'), (None, 'O'), (None, 'O'),
+                    (None, 'O'), (None, 'O'), (None, 'O')]
+
+        extractor = BioSeqTaggingExtractor(config)
+
+        for pack in pipeline.process_dataset(self.dataset_path):
+            for instance in pack.get(Sentence):
+                extractor.update_vocab(pack, instance)
+
+        for pack in pipeline.process_dataset(self.dataset_path):
+            for instance in pack.get(Sentence):
```

Comment:
I originally thought `Sentence` will be a config to the `extractor`. Looks like we have to specify it outside, which design do you think is better?

Reply:
`Sentence` should be passed in by Trainpreprocessor. Since I am writing the test code here now. I need to act like a train preprocessor to pass in this config.

Reply:
I think both of them works. It is a design choice. Since Most of my implementation is based on pack and instance. I think I will stick with this choice first.